src/Features/CSharp/Portable/UseRecursivePatterns/CSharpUseRecursivePatternsCodeRefactoringProvider.AnalyzedNode.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseRecursivePatterns
+{
+    internal sealed partial class CSharpUseRecursivePatternsCodeRefactoringProvider
+    {
+        private abstract class AnalyzedNode : Visitor<AnalyzedNode>
+        {
+            public abstract T Accept<T>(Visitor<T> visitor);
+            public abstract override string ToString();
+
+            public sealed override AnalyzedNode VisitSourcePattern(SourcePattern node) => throw ExceptionUtilities.Unreachable;
+        }
+
+        private sealed class Conjuction : AnalyzedNode
```

Comment:
Conjuntion junction, what's your function??

Reply:
simply a combination of nodes, could be patterns or expressions. later we try to simplify the final node in the Reducer.

Reply:
Sorry, bad joke: https://www.youtube.com/watch?v=RPoBE-E8VOc